src/main/scala/io/citrine/lolo/transformers/Standardizer.scala

```diff
@@ -0,0 +1,166 @@
+  override def getModel(): Model[PredictionResult[Any]] = new StandardizerModel(baseTrainingResult.getModel(), trans)
+
+  /**
+    * Get the hyperparameters used to train this model
+    *
+    * @return hypers set for model
+    */
+  override def getHypers(): Map[String, Any] = hypers
+
+  override def getFeatureImportance(): Option[Vector[Double]] = baseTrainingResult.getFeatureImportance()
+}
+
+class StandardizerModel[T](baseModel: Model[PredictionResult[T]], trans: Seq[Option[(Double, Double)]]) extends Model[PredictionResult[T]] {
+  /**
+    * Apply the model to a seq of inputs
+    *
+    * @param inputs to apply the model to
+    * @return a predictionresult which includes, at least, the expected outputs
+    */
+  override def transform(inputs: Seq[Vector[Any]]): StandardizerPrediction[T] = {
+    val standardInputs = Standardizer.applyStandardization(inputs, trans.tail)
+    new StandardizerPrediction(baseModel.transform(standardInputs), trans)
+  }
+}
+
+class StandardizerPrediction[T](baseResult: PredictionResult[T], trans: Seq[Option[(Double, Double)]]) extends PredictionResult[T] {
+  /**
+    * Get the expected values for this prediction
+    *
+    * @return expected value of each prediction
+    */
+  override def getExpected(): Seq[T] = {
+    baseResult.getExpected().map {
+      case x: Double => x * rescale + intercept
+      case x: Any => x
+    }.asInstanceOf[Seq[T]]
+  }
+
+  /**
+    * Get the uncertainty of the prediction
+    *
+    * For example, in regression this is sqrt(bias^2 + variance)
+    * @return uncertainty of each prediction
+    */
+  override def getUncertainty(): Option[Seq[Any]] = {
+    baseResult.getUncertainty() match {
+      case None => None
+      case Some(x: Seq[Double]) => Some(x.map(_ * rescale))
+      case Some(x: Seq[Any]) => Some(x)
+    }
+  }
+
+
+  /**
+    * Get the gradient or sensitivity of each prediction
+    *
+    * @return a vector of doubles for each prediction
+    */
+  override def getGradient(): Option[Seq[Vector[Double]]] = {
+    baseResult.getGradient() match {
+      case None => None
+      case Some(x) =>
+        Some(x.map(g => g.zip(trans.tail).map {
+          // If there was a (linear) transformer used on that input, take the slope "m" and rescale by it
+          case (y: Double, Some((_, m))) => y * rescale * m
+          // Otherwise, just rescale by the output transformer
+          case (y, None) => y * rescale
+        }))
+    }
+  }
+
+  val rescale = 1.0 / trans.head.map(_._2).getOrElse(1.0)
+  val intercept = trans.head.map(_._1).getOrElse(1.0)
+}
+
+object Standardizer {
+  def getStandardization(values: Seq[Double]): (Double, Double) = {
+    val mean = values.sum / values.size
+    val scale = Math.sqrt(values.map(v => Math.pow(v - mean, 2)).sum / values.size)
+    (mean, 1.0 / scale)
+  }
+
+  /**
+    * Get standardization for multiple values in a vector.
+    *
+    * This has a different name because scala erases the inner type of Seq[T]
```

Comment:
Pedantics alert! Isn't type erasure technically a JVM "feature", not scala behavior?